src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php

```diff
@@ -118,4 +223,211 @@ public function iShouldBeNotifiedThatItHasBeenSuccessfullyCreated(): void
+            'Product could not be edited'
+        );
+    }
+
+    /**
+     * @Then I should be notified that this product is in use and cannot be deleted
+     */
+    public function iShouldBeNotifiedThatThisProductIsInUseAndCannotBeDeleted(): void
+    {
+        Assert::false(
+            $this->responseChecker->isDeletionSuccessful($this->client->getLastResponse()),
+            'Product can be deleted, but it should not'
+        );
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully deleted
+     */
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyDeleted(): void
+    {
+        Assert::true(
+            $this->responseChecker->isDeletionSuccessful($this->client->getLastResponse()),
+            'Product still exists, but it should not'
+        );
+    }
+
+    /**
+     * @Then I should see :numberOfProducts products in the list
+     */
+    public function iShouldSeeProductsInTheList(int $count): void
+    {
+        Assert::count($this->responseChecker->getCollection($this->client->getLastResponse()), $count);
+    }
+
+    /**
+     * @Then I should( still) see a product with :field :value
+     */
+    public function iShouldSeeProductWith(string $field, string $value): void
+    {
+        Assert::true(
+            $this
+                ->responseChecker
+                ->hasItemWithTranslation($this->client->getLastResponse(), 'en_US', $field, $value)
+            ,
+            sprintf('Product has not %s with %s', $field, $value)
+        );
+    }
+
+    /**
+     * @Then I should not see any product with :field :value
+     */
+    public function iShouldNotSeeAnyProductWith(string $field, string $value): void
+    {
+        $this->client->update();
+
+        Assert::false(
+            $this
+                ->responseChecker
+                ->hasItemWithTranslation($this->client->getLastResponse(), 'en_US', $field, $value)
+            ,
+            sprintf('Product with %s set as %s still exists, but it should not', $field, $value)
+        );
+    }
+
+    /**
+     * @Then I should not be able to edit its code
+     */
+    public function iShouldNotBeAbleToEditItsCode(): void
+    {
+        $this->client->addRequestData('code', '_NEW');
+        $response = $this->client->index();
+
+        Assert::false(
+            $this->responseChecker->hasItemOnPositionWithValue(
+                $response,
+                0,
+                'code',
+                '/new-api/produ-apicts/_NEW'
+            ),
+            sprintf('It was possible to change %s', '_NEW')
+        );
+    }
+
+    /**
+     * @Then /^(this product) name should be "([^"]+)"$/
+     */
+    public function thisProductNameShouldBe(ProductInterface $product, string $name): void
+    {
+        $response = $this->client->index();
+
+        Assert::true(
+            $this->responseChecker->hasItemWithTranslation($response, 'en_US', 'name', $name),
```

Comment:
I have a feeling that this test is not testing exactly what we want to test. We are testing, that there is an item with such a name, while IMHO it would be better to open product show page and check it there. We can convert (this product) to product